components/TopupPage/TopupModal.tsx

```diff
@@ -1,42 +1,98 @@
-import { TOPUP_CROSSMINT_COLLECTION_ID } from "@/lib/consts";
 import { useSmartWalletProvider } from "@/providers/SmartWalletProvider";
 import { useTopupProvider } from "@/providers/TopupProvider";
-import { useUserProvider } from "@/providers/UserProvider";
-import { CrossmintEmbeddedCheckout } from "@crossmint/client-sdk-react-ui";
-import { Address } from "viem";
+import { Copy, Check } from "lucide-react";
+import { useState } from "react";
+import { Button } from "@/components/ui/button";
+import getViemNetwork from "@/lib/viem/getViemNetwork";
+import { CHAIN_ID } from "@/lib/consts";
 
 interface TopupModalProps {
   onClose: () => void;
 }
 
 export default function TopupModal({ onClose }: TopupModalProps) {
   const { smartWallet } = useSmartWalletProvider();
-  const { email } = useUserProvider();
-  const { callData } = useTopupProvider();
+  const { currency, depositAmount } = useTopupProvider();
+  const [copied, setCopied] = useState(false);
+
+  const handleCopy = async () => {
+    try {
+      await navigator.clipboard.writeText(smartWallet);
+      setCopied(true);
+      setTimeout(() => setCopied(false), 2000);
+    } catch (err) {
+      console.error("Failed to copy:", err);
+    }
+  };
```

Comment:
DRY principle
- actual: new copy code when we already have existing copy code in other parts of the app
- required: use shared copy code libs / hooks